tests/models/test_renderedtifields.py

```diff
@@ -0,0 +1,221 @@
+
+import unittest
+from datetime import date, timedelta
+
+from parameterized import parameterized
+
+from airflow import settings
+from airflow.models import Variable
+from airflow.models.dag import DAG
+from airflow.models.renderedtifields import RenderedTaskInstanceFields as RTIF
+from airflow.models.taskinstance import TaskInstance as TI
+from airflow.operators.bash import BashOperator
+from airflow.utils.session import create_session
+from airflow.utils.timezone import datetime
+
+TEST_DAG = DAG("example_rendered_ti_field", schedule_interval=None)
+START_DATE = datetime(2018, 1, 1)
+EXECUTION_DATE = datetime(2019, 1, 1)
+
+
+def clear_rendered_ti_fields():
+    with create_session() as session:
+        session.query(RTIF).delete()
+
+
+class ClassWithCustomAttributes:
+    """Class for testing purpose: allows to create objects with custom attributes in one single statement."""
+
+    def __init__(self, **kwargs):
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def __str__(self):
+        return "{}({})".format(ClassWithCustomAttributes.__name__, str(self.__dict__))
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+
+class TestRenderedTaskInstanceFields(unittest.TestCase):
+    """Unit tests for RenderedTaskInstanceFields."""
+
+    def setUp(self):
+        clear_rendered_ti_fields()
+
+    def tearDown(self):
+        clear_rendered_ti_fields()
+
+    @parameterized.expand([
+        (None, None),
+        ([], []),
+        ({}, {}),
+        ("test-string", "test-string"),
+        ({"foo": "bar"}, {"foo": "bar"}),
+        ("{{ task.task_id }}", "test"),
+        (date(2018, 12, 6), "2018-12-06"),
+        (datetime(2018, 12, 6, 10, 55), "2018-12-06 10:55:00+00:00"),
+        (
+            ClassWithCustomAttributes(
+                att1="{{ task.task_id }}", att2="{{ task.task_id }}", template_fields=["att1"]),
+            str(ClassWithCustomAttributes(
+                att1="test", att2="{{ task.task_id }}", template_fields=["att1"])),
+        ),
+        (
+            ClassWithCustomAttributes(nested1=ClassWithCustomAttributes(att1="{{ task.task_id }}",
+                                                                        att2="{{ task.task_id }}",
+                                                                        template_fields=["att1"]),
+                                      nested2=ClassWithCustomAttributes(att3="{{ task.task_id }}",
+                                                                        att4="{{ task.task_id }}",
+                                                                        template_fields=["att3"]),
+                                      template_fields=["nested1"]),
+            str(ClassWithCustomAttributes(nested1=ClassWithCustomAttributes(att1="test",
```

Comment:
Can you hard-code this string please? This is otherwise not really testing that the output is what we expect.

Reply:
Updated